### ISSUE TITLE
Updating the updater module

### ISSUE THREAD

**Issue by johnroper100** [CGCookie/blender-addon-updater#11]
Currently, when changes are made to `addon_updater.py` and or `addon_updater_ops.py`, the developer of each addon that it is used in has to manually updates the files. Is there a way that the updater could update it's own files? There may be some problems if a developer wants to customize the code though...

**Comment by carter2422** [comment]
Initially @TheDuckCow and I discussed whether or not we could release this as a Python module through PIP, but we haven't re-examined the issue in a while so I'm not sure. This is something I'd like to visit after the initial release.

**Comment by TheDuckCow** [comment]
Yes we could revisit PIP after this first release, it's making more sense to me now than before. There's still the oddity of whether it should update their ops code or just the module itself since, just due to complexity, most people are likely to use the example ops file and just slightly modify to fit their case. Eithe way, it would need to be the decision of the developer as we cannot anticipate what kind of changes they make and typically the updater and updater_ops will update together.

**Comment by TheDuckCow** [comment]
Note to self around getting this included as a pip module. A [note method](https://packaging.python.org/distributing/) and [reference setup](https://github.com/johnroper100/Blended/blob/master/setup.py).

**Comment by carter2422** [comment]
Getting this released through PIP would make me happy, mostly for the ability for developers to lock in the version or update at most any time.

**Comment by Lauloque** [comment]
Refreshing the discussion, wouldn't it be possible to use the demo addon as an "installer" addon, and have it install the addon-updater module in Blender's module folder?  

- **Windows:** `%USERPROFILE%\AppData\Roaming\Blender Foundation\Blender\3.6\scripts\modules`
- **macOS:** `/Users/$USER/Library/Application Support/Blender/3.6/scripts/modules`
- **Linux:** `$HOME/.config/blender/3.6/scripts/modules`

This way, it would also mean that addons developers do not have to ship a copy of addon-updater.
